source/mfs-painters/multiframepainter/KernelGenerationStage.cpp

```diff
@@ -89,11 +89,16 @@ void KernelGenerationStage::process()
     ssaoSamples = glkernel::kernel3{ static_cast<uint16_t>(s_ssaoKernelSize) };
     glkernel::sample::best_candidate(ssaoSamples);
     glkernel::scale::range(ssaoSamples, -1.0f, 1.0f);
-    for (auto& elem : ssaoSamples)
+    for (int i = 0; i < ssaoSamples.size(); i++)
     {
+        auto& elem = ssaoSamples[i];
         elem.z = glm::abs(elem.z);
         elem.z = std::max(0.1f, elem.z);
         elem = glm::normalize(elem);
+        // taken from http://john-chapman-graphics.blogspot.de/2013/01/ssao-tutorial.html
+        float scale = float(i) / float(ssaoSamples.size());
+        scale = glm::mix(0.1f, 1.0f, scale * scale);
+        ssaoSamples[i] *= scale;
     }
```

Comment:
without this change: ![2016-04-25 11_41_57-mfs-viewerd](https://cloud.githubusercontent.com/assets/1891915/14780241/618b691c-0adc-11e6-9d3f-690e880fb4a8.png)

with this change:
![2016-04-25 11_49_35-c__projects_multiframesampling_build_debug_mfs-viewerd exe](https://cloud.githubusercontent.com/assets/1891915/14780243/6980bdca-0adc-11e6-83b8-f858aeffaada.png)

you want more samples towards the origin because near geometry occludes more than distant geometry.

the mix probably has a similar function than our `std::max(0.1f, elem.z)`, to avoid samples too near to the origin. i think offsetting z is better and it actually looks better without the mix, i'll remove it.